core/services/llo/mercurytransmitter/transmitter.go

```diff
@@ -214,7 +237,14 @@ func (mt *transmitter) Close() error {
 			closers = append(closers, s.pm)
 			closers = append(closers, s.c)
 		}
-		return services.CloseAll(closers...)
+		if err := services.CloseAll(closers...); err != nil {
+			return err
+		}
+		// Unregister all the gauge funcs
+		for _, c := range mt.collectors {
```

Comment:
would it make sense to 

1.  Unregister collectors before services.CloseAll . This will ensure that even if services.CloseAll returns an error the Collectors will be unregistered.

2. If Unregister returns false, then at least log this